### PR TITLE
Refactoring browser component

### PR DESCRIPTION
**Pull request by yl-coder**
To improve browser testing by passing in browserConfig to the browserManager constructor.

### CODE REVIEW

**Comment by yl-coder** [comment]
I'm not sure having a browserManagerUtil is a good idea? The idea is to improve testability without using Reflection

**Comment by yl-coder** [comment]
Open for comments to refactor other areas of the browser component

**Comment by damithc** [comment]
Add values to the main Config class rather than create a new config class just for the browser? Later on we can use interfaces to ensure the Browser component sees only the values it needs.

**Comment by yl-coder** [comment]
Ok sure. Removed BrowserConfig

File: src/main/java/address/controller/MainController.java [hunk 1]
```diff
@@ -76,7 +76,8 @@ public MainController(MainApp mainApp, ModelManager modelManager, Config config)
         this.modelManager = modelManager;
         this.config = config;
         this.personList = modelManager.getAllViewablePersonsReadOnly();
-        this.browserManager = new BrowserManager(personList);
+        this.browserManager = new BrowserManager(personList,
+                                                 config.getBrowserNoOfPages());
```
Comment (damithc) [inline src/main/java/address/controller/MainController.java]: Doesn't look like you need a line break here.

File: src/main/java/address/util/Config.java [hunk 2]
```diff
@@ -25,6 +26,7 @@
     public Level currentLogLevel = DEFAULT_LOGGING_LEVEL;
     public HashMap<String, Level> specialLogLevels = DEFAULT_SPECIAL_LOG_LEVELS;
     private File prefsFileLocation = new File("preferences.json"); //Default user preferences file
+    public int browserNoOfPages = DEFAULT_BROWSER_NO_OF_PAGES;
```
Comment (damithc) [inline src/main/java/address/util/Config.java]: You can also add browser type (webview, jxbrowser) as a config parameter.

File: src/main/java/address/util/UrlUtil.java [hunk 3]
```diff
@@ -19,4 +20,33 @@ public static boolean compareBaseUrls(URL url1, URL url2) {
...
+    /**
+     * Gets a list of URL that may be potentially be used in the future.
+     * The {@link URL} that will be used at the current moment must be in the param urls. And it will be to determine
+     * which URL may be used in the future.
+     * @param urls The list of URL that contains URL that may be potentially used in the future.
+     * @param selectedUrl The index of the current URL in the list(urls) that will be used at this moment.
+     * @param noOfFuturisticUrl The number of future URLs you wish to obtain from the list.
+     * @return A list of url that may be potentially used in the future. Range of list size: [0, noOfFuturisticUrl]
+     */
+    public static List<URL> getFuturisticUrls(List<URL> urls, int selectedUrl, int noOfFuturisticUrl) {
```
Comment (damithc) [inline src/main/java/address/util/UrlUtil.java]: I don't think `futuristic` is the right term here.

**Comment by damithc** [comment]
No more comments for this version.

File: src/main/java/address/controller/MainController.java (hunk 1 above)
Comment (yl-coder) [inline src/main/java/address/controller/MainController.java]: Ok, fixed

File: src/main/java/address/util/UrlUtil.java (hunk 3 above)
Comment (yl-coder) [inline src/main/java/address/util/UrlUtil.java]: ok changed to future

File: src/main/java/address/util/Config.java (hunk 2 above)
Comment (yl-coder) [inline src/main/java/address/util/Config.java]: Noted. Added

File: src/main/java/address/util/UrlUtil.java [hunk 4]
```diff
@@ -16,7 +17,36 @@ public static boolean compareBaseUrls(URL url1, URL url2) {
         boolean isSameBaseUrl = url1.getHost().toLowerCase().replaceFirst("www.", "")
                                 .equals(url2.getHost().replaceFirst("www.", "").toLowerCase())
                                 && url1.getPath().replaceAll("/", "").toLowerCase()
-                                   .equals(url2.getPath().replaceAll("/", "").toLowerCase());
+                                .equals(url2.getPath().replaceAll("/", "").toLowerCase());
```
Comment (damithc) [inline src/main/java/address/util/UrlUtil.java]: URL class might have methods that could make this comparison simpler. Try googling for a better solution.

**Pull request merged**